components/modals/addListingModal.js

```diff
@@ -0,0 +1,152 @@
+import { doGetCategories } from '@api/category';
+
+const AddListing = () => {
+  const [open, setOpen] = useState(false);
+  const [categories, setCategories] = useState(["Yeezy", "KSubi", "Mike", "Kelly", "Kyle"]);
+  const [selectedCategory, setSelectedCategory] = useState("");
+  const [name, setName] = useState("");
+  const [description, setDescription] = useState("");
+  const [zip, setZip] = useState("");
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  // useEffect(async () => {
+  //   const data = await doGetCategories();
+  //   console.log("categories here", data);
+  //   setCategories(data);
+  // }, [])
+
+  const [dropped, dropdownOpen] = useState(false);
+
+  const handleDropdownOpen = () => {
+    dropdownOpen(true);
+  };
+
+  const handleDropdownClose = () => {
+    dropdownOpen(false);
+  };
+
+
+  // const handleListing = () => {
+
+  // }
+
+  return (
+    <>
+      <Button onClick={handleOpen}>Open modal</Button>
+      <Modal
+        open={open}
+        aria-labelledby='modal-modal-title'
+        aria-describedby='modal-modal-description'
+      >
+        <Box sx={{ 
+          width: 600,
+          height: 500, 
+          border: '1px grey',
+          backgroundColor: 'white',
+          m: 'auto',
+          mt: 30
+        }}>
+          
+          <TextDiv>
+            <TextField 
+              id="listing-name" 
+              sx={{ width: 500, m: 1 }} 
+              inputProps={{ maxLength: 50 }} 
+              label="Item/Service" 
+              variant="outlined" 
+              onChange={(e) => {
+                setName(e.target.value); 
+                console.log(name);}} 
+            />
+            <TextField 
+              id="listing-description" 
+              sx={{ width: 500, m: 1 }} 
+              inputProps={{ maxLength: 300 }} 
+              label="Description" 
+              variant="outlined" 
+              multiline 
+              onChange={(e) => {
+                setDescription(e.target.value); 
+                console.log(description);}} 
+            />
+
+            <FormControl sx={{ m: 1, minWidth: 120 }}>
+              <InputLabel id="demo-controlled-open-select-label">
+                Categories
+              </InputLabel>
+              <Select
+                labelId="demo-controlled-open-select-label"
+                id="demo-controlled-open-select"
+                open={dropped}
+                onClose={handleDropdownClose}
+                onOpen={handleDropdownOpen}
+                value={selectedCategory}
+                label="category"
+                onChange={(e) => {
+                  setSelectedCategory(e.target.value); 
+                  console.log(selectedCategory);}}
+                variant="outlined"
+              >
+                {categories.map((c) => {
+                  return <MenuItem value={c}>{c}</MenuItem>;
```

Comment:
for good practice's sake, would you be able to change this to:
`categories.map( (category, i) => {`
`  return <Component category={category} key={i} />`
`}`

Reply:
we'll get `console.error`s in the console if we map components without passing them a key prop